src/components/figures/chart/chartJs/chartjs_colorscale_plugin.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  CHART_AXIS_TITLE_FONT_SIZE,
+  CHART_COLORSCALE_WIDTH,
+  CHART_PADDING,
+} from "@odoo/o-spreadsheet-engine/constants";
+import { ChartType, Plugin } from "chart.js";
+import { Color, Locale } from "../../../..";
+import { getDefaultContextFont, humanizeNumber } from "../../../../helpers";
+
+export interface ChartColorScalePluginOptions {
+  position: "left" | "right" | "none";
+  colorScale: Color[];
+  fontColor?: Color;
+  minValue: number;
+  maxValue: number;
+  locale: Locale;
+}
+
+declare module "chart.js" {
+  interface PluginOptionsByType<TType extends ChartType> {
+    chartColorScalePlugin?: ChartColorScalePluginOptions;
+  }
+}
+
+/** This is a chartJS plugin that will draw the values of each data next to the point/bar/pie slice */
+export const chartColorScalePlugin: Plugin = {
+  id: "chartColorScalePlugin",
+  afterDatasetsDraw(chart: any, args, options: ChartColorScalePluginOptions) {
+    if (!options.position || options.position === "none" || !options.colorScale.length) {
+      return;
+    }
+    const ctx = chart.ctx as CanvasRenderingContext2D;
+    ctx.save();
+
+    ctx.textAlign = "center";
+    ctx.textBaseline = "middle";
+    ctx.miterLimit = 1; // Avoid sharp artifacts on strokeText
```

Comment:
👀 I find this quite interesting. do you think it could be useful outside of the chart drawings?  like for the borders maybe?